### ISSUE TITLE
Using "-Pagent" in Gradle with Graal 22.0.0-dev causes `java.lang.Error: The class ImageSingletons can only be used when building native images, i.e., when using the native-image command.`

### ISSUE THREAD

**Issue by GavinRay97** [graalvm/native-build-tools#186]
For a reproduction, run `gradle init`, and create a Java Application starter. Add the Graal gradle plugin.

Then, try to use the Polyglot API:
```java
import org.graalvm.polyglot.*;

class App {
    public static void main(String[] args) {
        Context polyglot = Context.create();
        Value array = polyglot.eval("js", "[1,2,42,4]");
        int result = array.getArrayElement(2).asInt();
        System.out.println(result);
    }
}
```

And run:
```sh
$ ./gradlew -Pagent run
```

The error I get is:
```ps1
Exception in thread "Thread-0" java.lang.Error: The class ImageSingletons can only be used when building native images, i.e., when using the native-image command.
        at org.graalvm.sdk/org.graalvm.nativeimage.impl.ImageSingletonsSupport.checkInstalled(ImageSingletonsSupport.java:69)
        at org.graalvm.sdk/org.graalvm.nativeimage.impl.ImageSingletonsSupport.get(ImageSingletonsSupport.java:63)
        at org.graalvm.sdk/org.graalvm.nativeimage.ImageSingletons.contains(ImageSingletons.java:107)
        at org.graalvm.sdk/org.graalvm.home.HomeFinder.getInstance(HomeFinder.java:98)
        at org.graalvm.locator/com.oracle.graalvm.locator.GraalVMLocator.getLanguagesLoader(GraalVMLocator.java:141)
        at org.graalvm.locator/com.oracle.graalvm.locator.GraalVMLocator.locate(GraalVMLocator.java:205)
        at org.graalvm.truffle/com.oracle.truffle.api.impl.TruffleLocator.loaders(TruffleLocator.java:76)
        at org.graalvm.truffle/com.oracle.truffle.polyglot.EngineAccessor.locatorLoaders(EngineAccessor.java:141)
        at org.graalvm.truffle/com.oracle.truffle.polyglot.EngineAccessor.locatorOrDefaultLoaders(EngineAccessor.java:160)
        at org.graalvm.truffle/com.oracle.truffle.polyglot.EngineAccessor$EngineImpl.loadServices(EngineAccessor.java:283)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadExternalDefaultProviders(LibraryFactory.java:439)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.getExternalDefaultProviders(LibraryFactory.java:430)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.initDefaultExports(LibraryFactory.java:221)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.<init>(LibraryFactory.java:216)
        at org.graalvm.truffle/com.oracle.truffle.api.library.DynamicDispatchLibraryGen.<init>(DynamicDispatchLibraryGen.java:31)
        at org.graalvm.truffle/com.oracle.truffle.api.library.DynamicDispatchLibraryGen.<clinit>(DynamicDispatchLibraryGen.java:23)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadGeneratedClass(LibraryFactory.java:777)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolveImpl(LibraryFactory.java:740)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolve(LibraryFactory.java:733)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.<init>(LibraryFactory.java:210)
        at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibraryGen.<init>(InteropLibraryGen.java:170)
        at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibraryGen.<clinit>(InteropLibraryGen.java:161)
        at java.base/java.lang.Class.forName0(Native Method)
        at java.base/java.lang.Class.forName(Class.java:467)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.loadGeneratedClass(LibraryFactory.java:777)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolveImpl(LibraryFactory.java:740)
        at org.graalvm.truffle/com.oracle.truffle.api.library.LibraryFactory.resolve(LibraryFactory.java:733)
        at org.graalvm.truffle/com.oracle.truffle.api.interop.InteropLibrary.<clinit>(InteropLibrary.java:2843)
        at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotValueDispatch.<clinit>(PolyglotValueDispatch.java:155)
        at org.graalvm.truffle/com.oracle.truffle.polyglot.PolyglotImpl.initialize(PolyglotImpl.java:163)
        at org.graalvm.sdk/org.graalvm.polyglot.impl.AbstractPolyglotImpl.setConstructors(AbstractPolyglotImpl.java:266)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine$1.loadAndValidateProviders(Engine.java:917)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine$1.run(Engine.java:890)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine$1.run(Engine.java:864)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:318)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine.initEngineImpl(Engine.java:864)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine.access$000(Engine.java:110)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine$ImplHolder.<clinit>(Engine.java:136)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine.getImpl(Engine.java:360)
        at org.graalvm.sdk/org.graalvm.polyglot.Engine$Builder.build(Engine.java:616)
        at org.graalvm.sdk/org.graalvm.polyglot.Context$Builder.build(Context.java:1822)
        at org.graalvm.sdk/org.graalvm.polyglot.Context.create(Context.java:971)
        at io.hasura.graal.server.App$SocketHandler.run(App.java:54)
```

Full `build.gradle` configuration is:
```gradle
plugins {
    id 'application'
    id 'org.graalvm.buildtools.native' version '0.9.9'
}

repositories {
    mavenCentral()
    gradlePluginPortal()
    maven {
        url "https://raw.githubusercontent.com/graalvm/native-build-tools/snapshots"
    }
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(17))
        vendor.set(JvmVendorSpec.matching("GraalVM Community"))
    }
}

graalvmNative {
     binaries.configureEach {
        agent {
            options.add("experimental-class-loader-support")
        }
    }
    binaries {
        main {
            debug = false
            verbose = true
            javaLauncher = javaToolchains.launcherFor {
                languageVersion = JavaLanguageVersion.of(17)
                vendor = JvmVendorSpec.matching("GraalVM Community")
            }
        }
    }
}

dependencies {
    implementation "org.graalvm:graal-sdk:0.33"
}

application {
    mainClass = 'io.hasura.graal.server.App'
}
```

**Comment by lazar-mitrovic** [comment]
`org.graalvm:graal-sdk:0.33` is a really old dependency (published in 2018 about 3.5 years ago). Current artifact is at:
```
implementation 'org.graalvm.sdk:graal-sdk:21.3.0'
```
For dev GraalVM builds you might get away with omitting this dependency entirely if you use javac from `${GRAALVM_HOME}/bin/javac` since newest sdk jar should then be on the classpath by default.

**Comment by GavinRay97** [comment]
Ah whoops. I googled `graal-sdk maven` and it lead me here, so I copy-pasted the version from this:
- https://github.com/oracle/graal/blob/master/sdk/README.md#getting-started

I probably should have double checked that 😅 
![image](https://user-images.githubusercontent.com/26604994/148276251-41e1c32e-461c-4b2e-ab41-f1f3b96c040d.png)

> For dev GraalVM builds you might get away with omitting this dependency entirely if you use javac from ${GRAALVM_HOME}/bin/javac since newest sdk jar should then be on the classpath by default.

This works -- but only if your code doesn't use the Graal namespaces like `native.c` or `polyglot`
If you try to compile with those in, it'll fail and say it can't find them =/

(This is using `graalvm-ce-java17-22.0.0-dev/bin/java` as the Gradle compiler executable)

```sh
Compiling with toolchain '/home/user/downloads/graalvm-ce-java17-22.0.0-dev'.
Starting process 'Gradle Worker Daemon 2'. Working directory: /home/user/.gradle/workers Command: /home/user/downloads/graalvm-ce-java17-22.0.0-dev/bin/java @/home/user/.gradle/.tmp/gradle-worker-classpath18025822024137698878txt -Xmx512m -Dfile.encoding=UTF-8 -Duser.country -Duser.language=en -Duser.variant worker.org.gradle.process.internal.worker.GradleWorkerMain 'Gradle Worker Daemon 2'
Successfully started process 'Gradle Worker Daemon 2'
Started Gradle worker daemon (0.192 secs) with fork options DaemonForkOptions{executable=/home/user/downloads/graalvm-ce-java17-22.0.0-dev/bin/java, minHeapSize=null, maxHeapSize=null, jvmArgs=[], keepAliveMode=SESSION}.
Compiling with JDK Java compiler API.
/home/user/projects/graalvm-native-lib-test/java/app/src/main/java/io/hasura/graal/server/App.java:15: error: package org.graalvm.polyglot does not exist
import org.graalvm.polyglot.Value;
                           ^
/home/user/projects/graalvm-native-lib-test/java/app/src/main/java/io/hasura/graal/server/App.java:16: error: package org.graalvm.polyglot does not exist
import org.graalvm.polyglot.Context;
                           ^
/home/user/projects/graalvm-native-lib-test/java/app/src/main/java/io/hasura/graal/server/App.java:22: error: cannot find symbol
    private static final Context context = Context.newBuilder("js").allowAllAccess(true).build();
```

With the following `build.gradle`:

```gradle
plugins {
    id 'application'
    id 'org.graalvm.buildtools.native' version '0.9.9'
}

repositories {
    mavenCentral()
    gradlePluginPortal()
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(17))
        vendor.set(JvmVendorSpec.matching("GraalVM Community"))
    }
}

ext {
    jacksonVersion = '2.13.1'
}

graalvmNative {
     binaries.configureEach {
        agent {
            options.add("experimental-class-loader-support")
        }
    }
    binaries {
        main {
            debug = false // Determines if debug info should be generated, defaults to false
            verbose = true // Add verbose output, defaults to false
            buildArgs.add('--language:js') 
            useFatJar = false // Instead of passing each jar individually, builds a fat jar
            javaLauncher = javaToolchains.launcherFor {
                languageVersion = JavaLanguageVersion.of(17)
                vendor = JvmVendorSpec.matching("GraalVM Community")
            }
        }
    }
}

dependencies {
    implementation "com.fasterxml.jackson.core:jackson-databind:$jacksonVersion"
}

application {
    mainClass = 'io.hasura.graal.server.App'
}
```
